### PR TITLE
Enable control flow guard for mono on windows

### PR DESCRIPTION
**Pull request by steveisok**
Resolves binskim warnings for the runtime and aot compilers on windows.

### CODE REVIEW

**Review by lewing — APPROVED** [review]

**Review by lambdageek — APPROVED** [review]

File: src/mono/CMakeLists.txt [hunk 1]
```diff
@@ -283,6 +283,7 @@ elseif(CLR_CMAKE_HOST_OS STREQUAL "windows")
     add_compile_options($<$<COMPILE_LANGUAGE:C,CXX>:/GF>) # enable string pooling
```
Comment (jkotas) [inline src/mono/CMakeLists.txt:283]: Nit: Control flow guard is not an optimization. I would enable it for all build flavors, not just for Release.

**Review by jkotas — APPROVED** [review]

**Review by jkotas — CHANGES_REQUESTED** [review]
Build breaks:
```
2024-04-19T19:57:55.6709922Z   MASM : fatal error A1000:cannot open file : $<0:/GUARD:cf
2024-04-19T19:57:55.6712646Z   [24/359] Building C object mono\eglib\CMakeFiles\eglib_objects.dir\D_\a\_work\1\s\src\native\minipal\utf8.c.obj
2024-04-19T19:57:55.8142203Z   cl : Command line warning D9024 : unrecognized source file type '$<1:/GUARD:cf', object file assumed
2024-04-19T19:57:55.8145990Z   cl : Command line warning D9027 : source file '$<1:/GUARD:cf' ignored
```

File: src/mono/CMakeLists.txt [hunk 2]
```diff
@@ -277,9 +277,9 @@ elseif(CLR_CMAKE_HOST_OS STREQUAL "windows")
   add_compile_options($<$<COMPILE_LANGUAGE:C,CXX>:/wd4324>)  # 'struct_name' : structure was padded due to __declspec(align())
   add_compile_options($<$<COMPILE_LANGUAGE:C,CXX>:/EHsc>)    # set exception handling behavior
   add_compile_options($<$<COMPILE_LANGUAGE:C,CXX>:/FC>)      # use full pathnames in diagnostics
-  add_compile_options($<$<COMPILE_LANGUAGE:C,CXX>:/GUARD:cf) # Enable control flow guard
+  append("/guard:cf" CMAKE_C_FLAGS CMAKE_CXX_FLAGS)          # Enable control flow guard
```
Comment (akoeplinger) [inline src/mono/CMakeLists.txt:280]: the reason why it didn't work was because you were missing the closing `>`, please change it back.

**Review by jkotas — APPROVED** [review]

**Comment by steveisok** [comment]
/ba-g The failure is similar to https://github.com/dotnet/runtime/issues/98578 except it's not appearing in the console log and so BA can't capture and relate the failure.

**Pull request merged**